aeson-compat.cabal

```diff
@@ -51,6 +50,8 @@ library
     , vector                   >=0.10 && <0.13
     , semigroups               >=0.16.2.2 && <0.19
     , tagged                   >=0.7.3 && <0.9
+  if impl(ghc < 7.10)
```

Comment:
I prefer `!impl(ghc >= 7.10)` which is semantically more correct (there *could* be other compilers than GHC)

Also made the same thing for `semigroups`